src/core/src/pso/graphics.rs

```diff
@@ -15,10 +15,19 @@
 //! Graphics pipeline descriptor.
 
 use {state as s};
-use Primitive;
 use super::EntryPoint;
-use super::input_assembler::{AttributeDesc, VertexBufferDesc};
-use super::output_merger::{ColorTargetDesc, DepthStencilDesc};
+use super::input_assembler::{AttributeDesc, InputAssemblerDesc, VertexBufferDesc};
+use super::output_merger::{ColorInfo, DepthStencilDesc};
+
+// Vulkan:
+//  - SpecializationInfo not provided per shader
+//  - TODO: infer rasterization discard from shaders?
```

Comment:
we absolutely could, yeah, but Vk portability would be a bit easier if we had this flag explicitly exposed